tools/code_format/check_format.py

```diff
@@ -175,6 +175,21 @@
 }
 # yapf: enable
 
+NON_TYPE_ALIAS_ALLOWED_TYPES = {
+    "^(?![A-Z].*).*$",
+    "^(.*::){,1}(StrictMock<|NiceMock<).*$",
+    "^(.*::){,1}(Test|Mock|Fake).*$",
+    "^Protobuf.*::.*$",
+    "^[A-Z]$",
+    "^.*, .*",
+    r"^.*\[\]$",
+}
+
+USING_TYPE_ALIAS_REGEX = re.compile("(using .* = .*;|typedef .* .*;)")
+SMART_PTR_REGEX = re.compile("std::(unique_ptr|shared_ptr)<(.*?)>(?!;)")
```

Comment:
Should we introduce type alias for `weak_ptr`? This kind of smart pointer is partly used in envoyproxy/envoy and envoy/envoy-wasm.

Reply:
Sure, but I'd recommend not expanding the scope of this series of PRs. Would be an easier follow-up, as there are relatively fewer weak_ptr uses I think.